pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2691,4 +2691,9 @@ self: super: {
   sdl2-gfx = overrideCabal { __propagatePkgConfigDepends = false; } super.sdl2-gfx;
   sdl2-ttf = overrideCabal { __onlyPropagateKnownPkgConfigModules = true; } super.sdl2-ttf;
 
+  # Needs git for compile-time insertion of commit hash into --version string.
+  kmonad = overrideCabal (drv: {
+    libraryToolDepends = (drv.libraryToolDepends or []) ++ [pkgs.git];
```

Comment:
Yes—well sort of. In case of a non-zero exit code (e.g., when `.git` is not present in the project root) then kmonad just puts an empty string there (c.f. [here](https://github.com/kmonad/kmonad/blob/master/src/KMonad/Args/TH.hs#L22)).

I guess the template Haskell could be a little bit smarter, either checking for the presence of the `git` executable, or the `.git` directory in the project root, and bailing out sooner when either is not found. That's something for 0.4.3, I guess :)

Reply:
(The `Paths_*` module generated by Cabal (for all its faults) does provide the version of the cabal package.)

Reply:
Makes sense!